labgrid/driver/usbvideodriver.py

```diff
@@ -54,7 +62,8 @@ def stream(self, caps_hint=None):
         tx_cmd = self.video.command_prefix + ["gst-launch-1.0", "-q"]
         tx_cmd += pipeline.format(self.video.path, caps).split()
         rx_cmd = ["gst-launch-1.0"]
-        rx_cmd += "fdsrc ! h264parse ! avdec_h264 ! glimagesink sync=false".split()
+        #rx_cmd += "fdsrc ! h264parse ! avdec_h264 ! glimagesink sync=false".split()
+        rx_cmd += "fdsrc ! jpegparse ! jpegdec ! glimagesink sync=false".split()
```

Comment:
This needs to be more flexible. Maybe return more information from _get_caps (encoder+decoder)?

Reply:
Yes, sgtm.

Reply:
I've talked with @michaelolbrich about this, the simpler approach is to use mkv and let the client use playbin.